app/models/concerns/media_archive_org_archiver.rb

```diff
@@ -29,19 +29,26 @@ def send_to_archive_org(url, key_id, _supported = nil)
         else
           data = snapshot_data.to_h.merge({ error: { message: "(#{body['status_ext']}) #{body['message']}", code: Lapis::ErrorCodes::const_get('ARCHIVER_ERROR') }})
           Media.notify_webhook_and_update_cache('archive_org', url, data, key_id)
-          if body['message']&.include?('The same snapshot') || body['status_ext'] == 'error:too-many-daily-captures'
-            PenderSentry.notify(
-              Pender::Exception::TooManyCaptures.new(body["message"]),
-              url: url,
-              response_body: body
-            )
+
+          if body['message']&.include?('The same snapshot') ||
+             body['status_ext'] == 'error:too-many-daily-captures' ||
+             body['status_ext'] == 'error:blocked-url'
+              PenderSentry.notify(
+                custom_exception(body['status_ext'], body["message"]),
+                url: url,
+                response_body: body
+              )
           else
             raise Pender::Exception::ArchiveOrgError, "(#{body['status_ext']}) #{body['message']}"
           end
         end
       end
     end
 
+    def custom_exception(status_ext, message)
+      status_ext == 'error:blocked-url' ?  Pender::Exception::BlockedUrl.new(message) : Pender::Exception::TooManyCaptures.new(message)
```

Comment:
I know that this method is only called for two cases, but in general I think it's more consistent and future-proof to return `Pender::Exception::TooManyCaptures` only if the error is `error:too-many-daily-captures`, instead of "any error that is not `error:blocked-url`". Something like:

```ruby
if status_ext == 'error:blocked-url'
  Pender::Exception::BlockedUrl.new(message)
elsif status_ext == 'error:too-many-daily-captures'
  Pender::Exception::TooManyCaptures.new(message)
else
  Pender::Exception::ArchiveOrgError.new(message)
end
```

Reply:
One of the cases dor `TooManyCaptures` is `body['message']&.include?('The same snapshot')`. In this case there is no `status_ext`. 

I can refactor so we take that in account, BUT I'm wondering: since @jayjay-w will have to look into the other cases we discussed in the ticket, would it be better to leave some duplication for now, and create a better abstraction then? So, leave it like this: https://github.com/meedan/pender/blob/752137c8cf828b9be30a6ace7c3dcd6b1b291c04/app/models/concerns/media_archive_org_archiver.rb#L33-L47

@caiosba what do you think?